pkg/apis/serving/v1alpha1/service_defaults.go

```diff
@@ -34,7 +34,7 @@ func (s *Service) SetDefaults(ctx context.Context) {
 		ans := s.GetAnnotations()
 		if ans == nil {
 			ans = map[string]string{}
-			defer s.SetAnnotations(ans)
+			s.SetAnnotations(ans)
```

Comment:
hmm, it is less obvious to me that this works as intended, since if `SetAnnotations()` made a defensive copy then the mutations below would not appear in the service's annotations.

Reply:
Hmm. So when reading this and seeing `defer SetAnnotations()` I assumed that `GetAnnotations()` made a defensive copy which led me to question why we didn't need to always call `SetAnnotations()`. This sent me down the path to this PR.

Given that the functions are just shorthand helper functions and the presence of `defer` makes it seem like it is necessary for correctness it felt more clear to remove.

I don't feel super strong about this so we can also discard this PR if you would like.